plugins/API/API.php

```diff
@@ -580,6 +582,59 @@ public function getSuggestedValuesForSegment($segmentName, $idSite)
 
         // if segment has suggested values callback then return result from it instead
         $suggestedValuesCallbackRequiresTable = false;
+
+        if (!empty($segment['suggestedValuesApi']) && is_string($segment['suggestedValuesApi']) && !Rules::isBrowserTriggerEnabled()) {
+            $now = Date::now()->setTimezone(Site::getTimezoneFor($idSite));
+            if (self::$_autoSuggestLookBack != 60) {
+                // in Auto suggest tests we need to assume now is in 2018...
+                // we do - 20 to make sure the year is still correct otherwise could end up being 2017-12-31 and the recorded visits are over several days in the tests we make sure to select the last day a visit was recorded
+                $now = $now->subDay(self::$_autoSuggestLookBack - 20);
+            }
+            // we want to avoid launching the archiver should browser archiving be enabled as this can be very slow... we then rather
+            // use the live api.
+            $period = 'year';
+            $date = $now->toString();
+            if ($now->toString('m') == '01') {
+                if (Rules::isArchivingDisabledFor(array($idSite), new Segment('', array($idSite)), 'range')) {
+                    $date = $now->subYear(1)->toString(); // use previous year data to avoid using range
+                } else {
+                    $period = 'range';
+                    $date = $now->subMonth(1)->toString() . ',' . $now->addDay(1)->toString();
+                }
+            }
+
+            $flat = strpos($segment['suggestedValuesApi'], 'Actions.') === 0;
```

Comment:
Wondering if there would be a better solution than hardcoding that for `Actions` only. Maybe we could always flatten the report if the API method has subtables with the same dimension or something like that 🤔

Reply:
I had it hard coded flattened initially, but then noticed there was an issue with the content dimensions for example where it then would return `content name - content piece` instead of just `content name`. technically, I suppose we could check the metadata and see if there is a subtableApiMethod defined and whether it is the same API method or a different one. Should there be a different api method we could disable flattening. Was hoping though to not needing to do it simply to keep things simple. I suppose it wouldn't be that complex though maybe. Will check later.

Reply:
Updated it @sgiehl